core/src/main/java/org/elasticsearch/index/query/GeoDistanceRangeQueryBuilder.java

```diff
@@ -111,65 +146,295 @@ public GeoDistanceRangeQueryBuilder includeLower(boolean includeLower) {
+        this.unit = unit;
+        return this;
+    }
+
+    public DistanceUnit unit() {
+        return unit;
+    }
+
     public GeoDistanceRangeQueryBuilder optimizeBbox(String optimizeBbox) {
         this.optimizeBbox = optimizeBbox;
         return this;
     }
 
+    public String optimizeBbox() {
+        return optimizeBbox;
+    }
+
     public GeoDistanceRangeQueryBuilder coerce(boolean coerce) {
         this.coerce = coerce;
         return this;
     }
 
+    public boolean coerce() {
+        return coerce;
+    }
+
     public GeoDistanceRangeQueryBuilder ignoreMalformed(boolean ignoreMalformed) {
         this.ignoreMalformed = ignoreMalformed;
         return this;
     }
 
+    public boolean ignoreMalformed() {
+        return ignoreMalformed;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException errors = new QueryValidationException();
+        if (fieldName == null) {
+            errors.addValidationError("fieldName cannot be null");
+        }
+        if (lat != null && lon == null || lon != null && lat == null) {
+            errors.addValidationError("Both lat and lon must be defined. Found lat [" + lat + "] and lon [" + lon + "]");
+        }
+        if ((lat == null && lon == null && geohash == null) || (geohash != null && (lat != null || lon != null))) {
+            errors.addValidationError("Must either define geohash or lat and lon fields. Found geohash [" + geohash + "], lat [" + lat
+                    + "] and lon [" + lon + "]");
+        }
+        if (from == null && to == null) {
+            errors.addValidationError("Must define at least one parameter from [from, to]");
+        }
+        if (from != null && !(from instanceof Number || from instanceof String)) {
+            errors.addValidationError("from must either be a number or a string. Found [" + from.getClass().getName() + "]");
+        }
+        if (to != null && !(to instanceof Number || to instanceof String)) {
+            errors.addValidationError("to must either be a number or a string. Found [" + to.getClass().getName() + "]");
+        }
+        if (optimizeBbox != null && !(optimizeBbox.equals("none") || optimizeBbox.equals("memory") || optimizeBbox.equals("indexed"))) {
+            errors.addValidationError("optimizeBbox must be one of [none, memory, indexed]");
+        }
+        return errors.validationErrors().isEmpty() ? null : errors;
+    }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        if (coerce) {
+            ignoreMalformed = true;
+        }
+
+        final boolean indexCreatedBeforeV2_0 = context.indexVersionCreated().before(Version.V_2_0_0);
+        GeoPoint point;
+        if (geohash == null) {
+            point = new GeoPoint(lat, lon);
+        } else {
+            point = new GeoPoint();
+            GeoHashUtils.decode(geohash, point);
```

Comment:
yep, its a bit of a weird method in that it both modifies the parameter and returns it